File: Python/guess_the_number.py
import random

random_number = random.randint(1, 100)
guesses = 0
print("I'm thinking of a number between 1 and 100.")
guess = int(input("Take a guess: "))
while True:
    guesses += 1
    if guess == random_number:
        print("You guessed it! The number was", random_number)
        print("You guessed it in", guesses, "attempts.")
        break
    elif guess < random_number:
        print("Your guess is too low.")
    else:
        print("Your guess is too high.")
    guess = int(input("Take a guess again: "))

File: C#/GuessTheNumber.cs
var randomNumber = new Random().NextInt64(1, 101);
var attempts = 0;

Console.WriteLine("I'm thinking of a number between 1 and 100.");
Console.Write("Take a guess: ");

while (true)
{
    var guess = int.Parse(Console.ReadLine());
    attempts++;

    if (guess == randomNumber)
    {
        Console.WriteLine($"You guessed it! The number was {randomNumber}.");
        Console.WriteLine($"You guessed it in {attempts} attempts.");
        break;
    }

    if (guess < randomNumber)
    {
        Console.WriteLine("Your guess is too low.");
    }
    else
    {
        Console.WriteLine("Your guess is too high.");
    }

    Console.Write("Take a guess again: ");
}